rosidl_generator_py/resource/_msg_support.c.em

```diff
@@ -309,6 +319,31 @@ nested_type = '__'.join(type_.namespaced_name())
       }
       rosidl_generator_c__String__assign(&dest[i], PyBytes_AS_STRING(encoded_item));
       Py_DECREF(encoded_item);
+@[    elif isinstance(member.type.value_type, AbstractWString)]@
+      assert(PyUnicode_Check(item));
+      // the returned string starts with a BOM mark
+      PyObject * encoded_item = PyUnicode_AsUTF16String(item);
+      if (!encoded_item) {
+        Py_DECREF(seq_field);
+        Py_DECREF(field);
+        return false;
+      }
+      char * buffer;
+      Py_ssize_t length;
+      int rc = PyBytes_AsStringAndSize(encoded_item, &buffer, &length);
+      if (rc) {
+        Py_DECREF(seq_field);
+        Py_DECREF(field);
+        return false;
+      }
+      // use offset of 2 to skip BOM mark
+      bool succeeded = rosidl_generator_c__U16String__assignn_from_char(&dest[i], buffer + 2, length - 2);
+      if (!succeeded) {
+        Py_DECREF(seq_field);
```

Comment:
Decref `encoded_item` (or move decref to be before L334).

Reply:
See https://github.com/ros2/rosidl_python/pull/47#discussion_r280238235